src/base/static/components/organisms/story-navigator.js

```diff
@@ -23,45 +26,48 @@ class StoryNavigator extends Component {
 
     this.state = {
       currentStory: Immutable.Map(),
-      currentRoute: "",
+      currentPlaceId: null,
       isStoryDataReady: false,
       isWithError: false,
     };
+
+    this.stories = [];
   }
 
   componentDidMount() {
-    try {
+    this.props.router.on("route", (fn, route) => {
+      const currentStoryState = this.getCurrentStoryState(route[1]);
+      currentStoryState && this.setState(currentStoryState);
+    });
+  }
```

Comment:
If you are going to add a subscription here, make sure to unsubscribe it as well in `componentWillUnmount` as well.